### DISCUSSION TITLE
How to make separate /var partition automounted?

### DISCUSSION THREAD

**Discussion by ragazenta** [systemd/mkosi#2304] [Q&A]
Hi,

I tried to separate `/var` partition, but it's not detected by gpt-auto-generator.

I add this file (`mkosi.repart/30-var.conf`)
```ini
[Partition]
Type=var
Format=ext4
CopyFiles=/var:/
Minimize=guess
```

I also put this file (`mkosi.extra/usr/lib/repart.d/30-var.conf`)
```ini
[Partition]
Type=var
Format=ext4
SizeMinBytes=2G
SizeMaxBytes=4G
Weight=125
```

I've successfully separated `/home` partition with above methods.

From Discoverable Partition Specifications, `/var` need special requirement.
> The first partition with this type UUID on the disk containing the root partition is automatically mounted to /var/ — under the condition that its partition UUID matches the first 128 bits of HMAC-SHA256(machine-id, 0x4d21b016b53445c2a9fb5c16e091fd2d) (i.e. the SHA256 HMAC hash of the binary type UUID keyed by the machine ID as read from /etc/machine-id. This special requirement is made because /var/ (unlike the other partition types listed here) is inherently private to a specific installation and cannot possibly be shared between multiple OS installations on the same disk, and thus should be bound to a specific instance of the OS, identified by its machine ID.

How to achieve that with `mkosi`?

Thanks.

**Comment by daandemeyer** [comment]
@ragazenta This is not an issue in mkosi, but one in systemd-repart, please open an issue in the systemd repository about this. AFAIK it's not possible today
